### PR TITLE
Add Python 2.7.12, Python 3.5.2, Pypy 5.3.1, Pip 8.1.2, Setuptools 23.1.0 and remove release candidate

### PR DESCRIPTION
**Pull request by cclauss**
https://www.python.org/downloads/release/python-2712/
https://www.python.org/downloads/release/python-352/
http://doc.pypy.org/en/latest/release-pypy2.7-v5.3.1.html
https://pypi.python.org/pypi/pip
https://pypi.python.org/pypi/setuptools

### CODE REVIEW

**Comment by kennethreitz** [comment]
already on my todo list for tomorrow!

**Comment by cclauss** [comment]
The future is now  ;-)  It is already tomorrow here is Switzerland!

Please keep up the _great_ work that you are doing!

**Comment by kennethreitz** [comment]
Fun fact: Larry cut this release from 10,000 feet in the air :)

**Comment by larryhastings** [comment]
Actually more like 30,000. When I first wrote that status update I had
10,000 on the brain because that's the leveling off altitude when they turn
on the WiFi.

**Comment by cclauss** [comment]
Once this is completed, does it make sense to update https://github.com/heroku/heroku-buildpack-python/blob/master/bin/compile#L34 to set 2.7.12 as the default Python and update pip to 8.1.2 and setuptools to 23.1.0?

**Comment by kennethreitz** [comment]
Yes and yes. 

I will be doing so first thing in the morning. Feel free to update the default, though, because I'll be merging this then.

File: bin/compile [hunk 1]
```diff
@@ -31,11 +31,11 @@ ENV_DIR=$3
...
 LEGACY_TRIGGER="lib/python2.7"
 
-DEFAULT_PYTHON_VERSION="python-2.7.11"
+DEFAULT_PYTHON_VERSION="python-2.7.12"
 DEFAULT_PYTHON_STACK="cedar-14"
 PYTHON_EXE="/app/.heroku/python/bin/python"
-PIP_VERSION="8.1.1"
-SETUPTOOLS_VERSION="20.4"
+PIP_VERSION="8.1.2"
+SETUPTOOLS_VERSION="23.1"
```
Comment (kennethreitz) [inline bin/compile]: 23.1.0

**Comment by kennethreitz** [comment]
:sparkles: :cake: :sparkles:

**Pull request merged**